projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/search-ai-input/search-ai-input.component.ts

```diff
@@ -0,0 +1,188 @@
+  MatTooltipModule
+} from '@angular/material/tooltip';
+import { ModalAiService } from '../../../../services/modal-ai.service';
+import { Agent } from '@alfresco/js-api';
+
+const MatTooltipOptions: MatTooltipDefaultOptions = {
+  ...MAT_TOOLTIP_DEFAULT_OPTIONS_FACTORY(),
+  disableTooltipInteractivity: true
+};
+
+@Component({
+  standalone: true,
+  imports: [
+    CommonModule,
+    TranslateModule,
+    MatButtonModule,
+    MatIconModule,
+    MatFormFieldModule,
+    MatInputModule,
+    A11yModule,
+    FormsModule,
+    ReactiveFormsModule,
+    MatSelectModule,
+    IconComponent,
+    AvatarComponent,
+    MatCardModule,
+    MatTooltipModule
+  ],
+  selector: 'aca-search-ai-input',
+  templateUrl: './search-ai-input.component.html',
+  styleUrls: ['./search-ai-input.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  providers: [{ provide: MAT_TOOLTIP_DEFAULT_OPTIONS, useValue: MatTooltipOptions }]
+})
+export class SearchAiInputComponent implements OnInit, OnDestroy {
+  @Input()
+  placeholder: string;
+
+  @Input()
+  agentId: string;
+
+  @Input()
+  useStoredNodes: boolean;
+
+  @Output()
+  searchSubmitted = new EventEmitter<void>();
+
+  private readonly storedNodesKey = 'knowledgeRetrievalNodes';
+
+  private _agentControl = new FormControl<Agent>(null);
+  private _agents: Agent[] = [];
+  private onDestroy$ = new Subject<void>();
+  private selectedNodesState: SelectionState;
+  private _queryControl = new FormControl('');
+  private _initialsByAgentId: { [key: string]: string } = {};
+
+  get agentControl(): FormControl<Agent> {
+    return this._agentControl;
+  }
+
+  get agents(): Agent[] {
+    return this._agents;
+  }
+
+  get queryControl(): FormControl<string> {
+    return this._queryControl;
+  }
+
+  get initialsByAgentId(): { [key: string]: string } {
+    return this._initialsByAgentId;
+  }
+
+  constructor(
+    private store: Store<AppStore>,
+    private searchAiService: SearchAiService,
+    private notificationService: NotificationService,
+    private agentService: AgentService,
+    private userPreferencesService: UserPreferencesService,
+    private translateService: TranslateService,
+    private modalAiService: ModalAiService
+  ) {}
+
+  ngOnInit(): void {
+    if (!this.useStoredNodes) {
+      this.store
+        .select(getAppSelection)
+        .pipe(takeUntil(this.onDestroy$))
+        .subscribe((selection) => {
+          this.selectedNodesState = selection;
+        });
+    } else {
+      this.selectedNodesState = JSON.parse(this.userPreferencesService.get(this.storedNodesKey));
+    }
+
+    this.agentService
+      .getAgents()
+      .pipe(takeUntil(this.onDestroy$))
+      .subscribe(
+        (agents) => {
+          // TODO remove mocked avatar images after backend is done (https://hyland.atlassian.net/browse/ACS-8769)
+          const images = ['assets/images/avatars/Blue.png', 'assets/images/avatars/Gold.png', 'assets/images/avatars/Pink.png'];
```

Comment:
Maybe this part can be moved to a util file or some other place? As I think it looks basically the same as the one in agents button component